lookout/style/format/quality_report.py

```diff
@@ -20,35 +54,312 @@ def quality_report(input_pattern: str, bblfsh: str, language: str, n_files: int,
     rules = model[language]
     print("Model parameters: %s" % rules.origin_config)
     print("Stats about rules: %s" % rules)
-
+    # prepare input files
     client = BblfshClient(bblfsh)
     filenames = glob.glob(input_pattern, recursive=True)
     files = prepare_files(filenames, client, language)
     print("Number of files: %s" % (len(files)))
-
+    # extract features and check result
     fe = FeatureExtractor(language=language, **rules.origin_config["feature_extractor"])
     res = fe.extract_features(files)
-
     if res is None:
         print("Failed to parse files, aborting report...")
         return
     X, y, vnodes_y, vnodes = res
-
+    # predict with model and generate report
     y_pred, _ = rules.predict(X, vnodes_y, vnodes, language)
+    target_names = [CLASSES[cls_ind] for cls_ind in unique_labels(y, y_pred)]
+    print(generate_report(y=y, y_pred=y_pred, target_names=target_names, vnodes_y=vnodes_y,
+                          n_files=n_files))
+
+
+class ReportModel(AnalyzerModel):
+    """Model for ReportAnalyzer - store configuration."""
+
+    NAME = "style.format.quality_report_analyzer.ReportModel"
+
+    def __init__(self, config=None, **kwargs):
+        """Initialize ReportModel with configuration."""
+        super().__init__(**kwargs)
+        self.config = config
+
+    def _generate_tree(self) -> dict:
+        """
+        Return the tree to store in ASDF file.
+
+        :return: dict.
+        """
+        return dict(config=[self.config])
+
+    def _load_tree(self, tree: dict) -> None:
+        """
+        Attach the needed data from the tree.
+
+        :param tree: asdf file tree.
+        :return: None
+        """
+        self.config = tree["config"]
+
+
+class ReportAnalyzer(FormatAnalyzer):
```

Comment:
Is it possible not to derive from `FormatAnalyzer`? I see that you need the same model, so you can fake it with `model = FormatModel`. The rest of the parent's methods seem to be not used.